OWML.ModHelper.Interaction/ModInteraction.cs

```diff
@@ -0,0 +1,72 @@
+﻿using OWML.Common;
+using OWML.ModHelper.Events;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace OWML.ModHelper.Interaction
+{
+    public class ModInteraction : IModInteraction
+    {
+        private readonly List<ModBehaviour> _modList = new List<ModBehaviour>();
+        private readonly IModFinder _finder;
+        private readonly Dictionary<IModData, List<IModData>> dependantDict = new Dictionary<IModData, List<IModData>>();
+
+        public ModInteraction(List<ModBehaviour> list, IModFinder finder)
+        {
+            _modList = list;
+            _finder = finder;
+
+            foreach (var mod in _finder.GetMods())
+            {
+                List<IModData> temp = new List<IModData>();
+                foreach (var mod2 in _finder.GetMods())
+                {
+                    if (mod2.Manifest.Dependencies != null && mod2.Manifest.Dependencies.Contains(mod.Manifest.Name))
+                    {
+                        temp.Add(mod2);
+                    }
+                }
+                dependantDict.Add(mod, temp);
+            }
+        }
+
+        public IList<IModData> GetMods()
+        {
+            return _finder.GetMods();
+        }
+
+        public IList<IModData> GetDependants(string dependencyUniqueName)
+        {
+            return dependantDict.Where(x => x.Key.Manifest.UniqueName == dependencyUniqueName).FirstOrDefault().Value;
```

Comment:
Can dependantDict.Where(...).FirstOrDefault() be null? If so this will crash

Reply:
Nope. dependantDict contains every mod, so it'll only be empty if there are no mods. And if there's no mods, there's nothing to call this :P

Reply:
If that's the case then you can use First instead of FirstOrDefault ;)